pkg/apis/kops/validation/validation.go

```diff
@@ -457,12 +457,32 @@ func validateNetworkingCilium(c *kops.ClusterSpec, v *kops.CiliumNetworkingSpec,
 		allErrs = append(allErrs, field.Forbidden(fldPath.Root().Child("spec", "kubeProxy", "enabled"), "When Cilium NodePort is enabled, kubeProxy must be disabled"))
 	}
 
-	if v.Ipam == kops.CiliumIpamEni {
-		if c.CloudProvider != string(kops.CloudProviderAWS) {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("ipam"), "Cilum ENI IPAM is supported only in AWS"))
-		}
-		if !v.DisableMasquerade {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("disableMasquerade"), "Masquerade must be disabled when ENI IPAM is used"))
+	if v.EnablePolicy != "" {
+		allErrs = append(allErrs, IsValidValue(fldPath.Child("enablePolicy"), &v.EnablePolicy, []string{"default", "always", "never"})...)
+	}
+
+	if v.Tunnel != "" {
+		allErrs = append(allErrs, IsValidValue(fldPath.Child("tunnel"), &v.Tunnel, []string{"vxlan", "geneve", "disabled"})...)
+	}
+
+	if v.MonitorAggregation != "" {
+		allErrs = append(allErrs, IsValidValue(fldPath.Child("monitorAggregation"), &v.MonitorAggregation, []string{"low", "medium", "maximum"})...)
+	}
+
+	if v.ContainerRuntimeLabels != "" {
+		allErrs = append(allErrs, IsValidValue(fldPath.Child("containerRuntimeLabels"), &v.ContainerRuntimeLabels, []string{"none", "containerd", "crio", "docker", "auto"})...)
```

Comment:
Should probably warn that this is deprecated when cilium version > 1.7.0 and that setting this has no effect. The field may be removed in the future.

Reply:
There is no mechanism for warnings that I am aware of.

Reply:
I wonder if you can just print. `upup/pkg/fi/cloudup/apply_cluster.go` does that in a couple of places and validation is called from there. Or that particular validation logic could be done there (seems unclean to me though)

Reply:
Seems hokey to me. Wouldn't work well with `kops edit cluster`. Since it would only be ignored, a warning would be low-value.

Reply:
It would be shown on `kops update cluster` too, I think. But no problem to leave it out